service/history/api/recordchildworkflowcompleted/api.go

```diff
@@ -153,6 +157,28 @@ func recordChildWorkflowCompleted(
 				return nil, consts.ErrChildExecutionNotFound
 			}
 
+			childrenInitializedAfterResetPoint := mutableState.GetChildrenInitializedPostResetPoint()
+			if len(childrenInitializedAfterResetPoint) > 0 {
+				// This parent was reset and it also has some children that potentially were restarted.
+				initiatedEvent, err := mutableState.GetChildExecutionInitiatedEvent(ctx, parentInitiatedID)
```

Comment:
hmm I don't think we need to load the initiatedEvent. Both child workflow type and child workflowID are available in ChildExecutionInfo in mutable state I think.